src/main/java/org/mule/extension/ftp/internal/command/FtpCopyCommand.java

```diff
@@ -71,11 +77,18 @@ protected void copyDirectory(FileConnectorConfig config, URI sourceUri, URI targ
         }
 
         FileAttributes fileAttributes = new FtpFileAttributes(createUri(sourceUri.getPath(), file.getName()), file);
+        String filePath = fileAttributes.getPath();
 
         URI targetFileUri = createUri(targetUri.getPath(), fileAttributes.getName());
         if (fileAttributes.isDirectory()) {
-          copyDirectory(config, createUri(fileAttributes.getPath()), targetFileUri, overwrite, writerConnection);
+          if (LOGGER.isTraceEnabled()) {
+            LOGGER.trace("Copy directory {} to {}", filePath, targetUri);
+          }
```

Comment:
This is a cheap interpolation because these fields have already been resolved. I don't think the if is necessary in this case. Same applies to line 90.